src/app/orders/order-detail/order-detail.page.ts

```diff
@@ -1,15 +1,148 @@
+import { ProfileModel } from 'src/providers/models/profile.model';
 import { Component, OnInit } from '@angular/core';
+import { OrderService } from '../../../providers/services/orders/order.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { StorageServiceProvider } from '../../../providers/services/storage/storage.service';
+import { CONSTANTS } from '../../../providers/utils/constants';
+import { WidgetUtilService } from '../../../providers/utils/widget';
 
 @Component({
   selector: 'app-order-detail',
   templateUrl: './order-detail.page.html',
   styleUrls: ['./order-detail.page.scss'],
 })
 export class OrderDetailPage implements OnInit {
+  orderItems: any = [];
+  orderDetail: any = {};
+  showImportOrder = false;
+  showLoader = false;
+  showCsvButton = false;
+  showCancelOrder = false;
+  orderItemsAvailable = false;
+  csvData: any[] = [];
+  loaderDownloading: any;
+  headerRow: any[] = [];
+  profile: ProfileModel;
 
-  constructor() { }
+
+  constructor(
+    private storageService: StorageServiceProvider,
+    private orderService: OrderService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private widgetUtil: WidgetUtilService
+  ) { }
 
   ngOnInit() {
+    this.checkData();
+    this.route.queryParams.subscribe((params) => {
```

Comment:
We should not extract queryParams from subscribe, extract it from snapshot.

Reply:
replace it to this.route.snapshot.queryParamMap.get('order');